uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1025.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1025 {
+
+  /**
+   * Deletes some demographics fields where they are not in use
+   *
+   * @return bool
+   */
+  public function upgrade_1025() {
+
+    $this->deleteExtendDemographicFields([
+      'Ethnicity',
+      'Religion',
+      'Sexual_Orientation',
+    ]);
+
+    return TRUE;
+  }
+
+  /**
+   * Deletes Custom Demographic Fields only if they are not used
+   *
+   * @param $fieldsToDelete
+   */
+  private function deleteExtendDemographicFields($fieldsToDelete) {
+    $customGroup = civicrm_api3('CustomGroup', 'get', [
+      'name' => ['LIKE' => 'Extended_Demographics'],
+    ]);
+    $customGroup = array_shift($customGroup['values']);
+    $customFields = civicrm_api3('CustomField', 'get', [
+      'name' => ['IN' => $fieldsToDelete],
+    ]);
+    foreach ($customFields['values'] as $customField) {
+      $query = 'SELECT * FROM ' . $customGroup['table_name'] . ' WHERE ' . $customField['column_name'] . ' NOT LIKE "%Not Applicable%" AND ' . $customField['column_name'] . ' IS NOT NULL AND ' . $customField['column_name'] . ' <> ""';
```

Comment:
This isn't very readable, how about using `sprintf` and also creating some variables instead of accessing by array every time, for example:

```
$tableName = $customGroup['table_name'];
$queryFormat = 'SELECT COUNT(id) FROM "%s"'
  . ' WHERE "%s" NOT LIKE "%Not Applicable%"'
  . ' AND "%s" IS NOT NULL'
  . ' AND "%s" <> ""';

foreach ($customFields['values'] as $customField) {
  $column = $customField['column_name'];
  $query = sprintf($queryFormat, $tableName, $column, $column, $column);
```

Creating variables helps keep the lines shorter and makes the code more readable.